src/test/java/wolox/reactortraining/repositories/UserRepositoryTest.java

```diff
@@ -0,0 +1,124 @@
+package wolox.reactortraining.repositories;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+import wolox.reactortraining.dtos.TopicDto;
+import wolox.reactortraining.dtos.UserDto;
+import wolox.reactortraining.models.Topic;
+import wolox.reactortraining.models.User;
+
+@RunWith(SpringRunner.class)
+@DataMongoTest
+public class UserRepositoryTest {
+
+    @Autowired
+    private TopicRepository topicRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Before
+    public void setup() {
+        userRepository
+            .deleteAll()
+            .block();
+    }
+
+    @Test
+    public void givenNoPersistedUsers_whenFindAllIsCalled_thenReturnEmptyResultSet() {
+        Flux<User> findAllFlux = userRepository.findAll();
+
+        StepVerifier
+            .create(findAllFlux)
+            .expectSubscription()
+            .expectComplete()
+            .verify();
+    }
+
+    @Test
+    public void givenPersistedUsersInDB_whenFindAllIsCalled_thenReturnNonEmptyResultSet() {
+        UserDto dto = new UserDto();
+        dto.setUsername(randomAlphanumeric(6));
```

Comment:
refactoring, this code is repeated several times in the class

Reply:
The principle that quotes, is about keeping the test independent, ie should only do one thing and not interfere with the other tests, but repeated code, refers to the discipline refactoring. https://refactoring.com/.